### ISSUE TITLE
Regression Function

### ISSUE THREAD

**Issue by eric-bueno** [broadinstitute/segmentation_experiments#24]
- similar pipeline to color classification
- Load composite of masks for each image to get the ground truth
- Extend crop function to get crop of image and mask composite
- With these crops, compute average size of nucleus
- Do not need to keep composites for training, just need to create the label
- Make sure code is robust; can it handle a blank crop (no nuclei in image)
- Neural network should work the same
- Except the Dense layer output will only be one: we just need the size
- Use absolute error loss function (or square error)
- Metric is error, not accuracy

**Comment by eric-bueno** [comment]
Progress has been made in commit cd09b749e73a0e23f17f2606c3dc74e6d3071214

**Comment by jccaicedo** [comment]
The next goal here can be the creation of a dataset that has a variety of object sizes, similar to what we did with sampling crops of different colors in a balanced way.

Before training a neural network, you can follow the next steps to create a balanced dataset for size training:

1. Compute the distribution of sizes in the training set. This can be histogram of average sizes in all images with 100 bins, for instance.
1. For each bin in the distribution, take 200 random images and rescale them to match the size of the bin. For instance, a hypothetical distribution of sizes with 3 bins may have objects of 100, 200, and 300 pixels of area. For each bin, sample 200 images at random and rescale them with target size 100, 200, and 300 respectively.
1. Using 100 bins, this will give you 20,000 examples for training. Prepare the images and the labels (size) for training the neural net.
1. Run experiments by training a neural net that learns to predict size from these 20,000 examples. It probably will take longer than before, but with a balanced set of sizes, we hope to observe better accuracy.

@eric-bueno  Feel free to expand this thread with questions or any results that you may get in the meantime!

**Comment by eric-bueno** [comment]
Progress has been made in commit e1570c168905553fb248164a50aaaa6a5002cb95

1. Rescaling notebook now has 20,000 cropped examples of rescaled images and masks. The average sizes of the rescaled images are within ~20 pixels of the expected target values.
2. I wrote a function which returns num_samples random crops of rescaled images (where the images are rescaled to match the bins of the distribution)
3. Currently attempting to fit the data into the model

**Comment by eric-bueno** [comment]
Progress has been made in commit c3fc191b23dbebaa32edf37ca0a57f51f15a97d0

1. Changed number of examples to 10,000 because 20,000 samples created memory issues
2. Regression model has achieved mean_absolute_error value of ~40 pixels, which is well within the size of bins making up the distribution
3. Cleaned up unneeded cells in 04_train_regression.ipynb
4. Next step is to further augment the data
